### ISSUE TITLE
Clean up deployment/installation process

### ISSUE THREAD

**Issue by bent0b0x** [Paratii-Video/paratii-lib#23]
**Problem**
- [ ] Currently, this library is not easy to install and use. It requires installing the `dev` branch of this repo, `npm link`-ing, and then manually running `npm run pack` to bundle the lib. 

- [ ] another issue is that when another module includes `paratii-lib` in its bundle via `webpack`, it includes `paratii.ipfs.uploader.js` (which should only run on node). This leads to an error that the `'fs'` module cannot be resolved. `paratii-lib` tries to handle this with [this line](https://github.com/Paratii-Video/paratii-lib/blob/dev/package.json#L6), but it does not seem to work w/ `webpack`.

**Proposed Solution(s) (once `dev` is merged into `master`)**
Either automatically bundle the lib on `postInstall`, or publish the lib to `npm` and handle bundling automatically via a `prePublish` hook

cc @ya7ya

**Comment by jellegerbrandy** [comment]
Third option: have developers create the bundle and commit the dist folder on github. (This is what we are doing now). 

I don't really get your first point - with this 3rd option, installing should "just work" (I mean, apart from bugs and such)

I personally think that including the build in the code repo is not very elegant  repo is not very elegant, and would vote for removing the dist dir from the repo and replacing it with a `postInstall`.

i vote against having adding a dependency on npmjs.com (publishing to npmjs is great, but the lib should be installable directly from the code on github as well)

**Comment by bent0b0x** [comment]
My first point is simply that the code on `master` does not have the correct bundle that is usable by other apps/libs. So you have to check out another branch, npm link, and run the webpack script locally rather than just adding this as a dependency to your `package.json`.  So yes the current infrastructure/plan should work, but it isn't right now :)

**Comment by jellegerbrandy** [comment]
Yes, true.


I think we should **pin** our versions from now on, so we always have working builds that refer to the intended version of hte packages. This is the current entry in package.json of paratii-lib: 

    "paratii-contracts": "github:Paratii-Video/paratii-contracts#ed7308ea956777f60db0c976bda6889666cd2df4",
  
This way, the code won't break if we upgrade the `paratii-contracts`. ,Once we start seriously versioning our code, the links would refer to the branch of the version:

    "paratii-contracts": "github:Paratii-Video/paratii-contracts#v1.3.2",

Similarly, we should pint the `paratii-lib` version when we use it in the portal...
